File: packages/recycleros_domain/lib/src/models/opportunity.dart
import 'procurement_intent.dart';
import 'opportunity_source.dart';
import 'opportunity_status.dart';

class Opportunity {
  final String opportunityId;
  final String opportunityCode;
  final String title;
  final OpportunitySource source;
  final OpportunityStatus status;
  final ProcurementIntent procurementIntent;
  final String? vin;
  final int? year;
  final String? make;
  final String? model;
  final double? estimatedMaxBid;
  final double? estimatedNetProfit;
  final double? confidenceScore;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Opportunity({
    required this.opportunityId,
    required this.opportunityCode,
    required this.title,
    required this.source,
    required this.status,
    required this.procurementIntent,
    this.vin,
    this.year,
    this.make,
    this.model,
    this.estimatedMaxBid,
    this.estimatedNetProfit,
    this.confidenceScore,
    required this.createdAt,
    required this.updatedAt,
  });
}

File: services/api/src/schemas/opportunity.py
from pydantic import BaseModel, Field

class OpportunityCreate(BaseModel):
    organization_id: str | None = None
    workspace_id: str | None = None
    title: str = Field(min_length=1)
    source_type: str = "manual"
    procurement_intent: str = "undecided"
    vin: str | None = None
    year: int | None = Field(default=None, ge=1886, le=2100)
    make: str | None = None
    model: str | None = None
    estimated_max_bid: float | None = Field(default=None, ge=0)
    estimated_net_profit: float | None = None
    confidence_score: float | None = Field(default=None, ge=0, le=100)
